.github/workflows/release-cycle.yml

```diff
@@ -0,0 +1,212 @@
+          git checkout -b $RELEASE_BRANCH
+      - name: Enter in prerelease state
+        run: |
+          npx changeset pre enter rc
+          git add .
+          git commit -m "Start release candidate"
+          git push --all origin
+
+  changesets_pr:
+    needs: check_state
+    name: Changesets PR
+    if: needs.check_state.outputs.changesets_pr == 'true'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # To get all tags
+      - name: Set up environment
+        uses: ./.github/actions/setup
+      - name: Set release title
+        env:
+          BRANCH_NAME: ${{ github.ref_name }}
+        run: |
+          BRANCH_VERSION=$(npx semver -c $(echo $BRANCH_NAME | sed "s/release-v//g"))
+          PACKAGE_JSON_NEXT_PATCH=$(npx semver -i $(node -p -e "require('./package.json').version"))
+          NEXT_VERSION=$(npx semver $BRANCH_VERSION $PACKAGE_JSON_NEXT_PATCH | tail -n 1)
+          echo "TITLE=Release v${NEXT_VERSION}" >> $GITHUB_ENV
+      - name: Create PR
+        uses: changesets/action@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          version: npm run version
+          title: ${{ env.TITLE }}
+          commit: ${{ env.TITLE }}
+          body: | # Wait for support on this
+            This is an automated PR for releasing ${{ github.repository }}
+            Check [CHANGELOG.md](${{ github.repository }}/CHANGELOG.md)
+
+  promote_to_final:
+    needs: check_state
+    environment: Release Cycle
+    name: Promote to final release
+    if: needs.check_state.outputs.promote_to_final == 'true'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.GH_TOKEN }} # Needed so release cycle triggers when pushing
+      - name: Set up environment
+        uses: ./.github/actions/setup
+      - run: bash scripts/git-user-config.sh
+      - name: Exit prerelease state if needed
+        run: |
+          PRERELEASE=$(node -p -e 'require("@changesets/pre").readPreState(process.cwd()).then(prestate => console.log(prestate?.mode == "pre"))' | tail -n 1)
+          if [ $PRERELEASE != "false" ]; then
+            npx changeset pre exit rc
+            git add .
+            git commit -m "Exit release candidate"
+            git push --all origin
+          fi
+
+  publish:
+    needs: check_state
+    environment: npm
+    name: Publish to npm
+    if: needs.check_state.outputs.publish == 'true'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up environment
+        uses: ./.github/actions/setup
+      - run: bash scripts/git-user-config.sh
+      - name: Tag
+        run: npx changeset tag
+      - name: Publish to npm
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+        run: |
+          cd contracts
+          echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > $HOME/.npmrc
+          npm publish
```

Comment:
Another problem here is that we have to publish the `contracts` subdirectory instead of the root.